### DISCUSSION TITLE
using runtimeConfig inside composables throws nuxt instance error

### DISCUSSION THREAD

**Discussion by shba007** [nuxt/nuxt#16357] [Questions]
Inside composables/api.ts

    import { $fetch } from "ofetch";
    
    const config = useRuntimeConfig()
    
    export const useAPI = $fetch.create({
      baseURL: config.public.apiURL,
      headers: { 'Authorization': `Bearer ${""}` },
      onRequest({ request, options }) {
        console.log("Intercepted API", request);
    
        options.headers = options.headers || {}
        // options.headers.authorization = '...'
      }
    })

throws nuxt intance error

**Comment by larzon83** [comment]
I think you need to put inside the function

**Reply by shba007** [reply]
If create a function like this

    import { $fetch } from "ofetch";
    
    export const useAPI = () => {
      const config = useRuntimeConfig()
    
      return $fetch.create({
        baseURL: config.public.apiURL,
        headers: { 'Authorization': `Bearer ${""}` },
        onRequest({ request, options }) {
          console.log("Intercepted API", request);
    
          options.headers = options.headers || {}
          // options.headers.authorization = '...'
        }
      })
    }

It says Cannot stringify a function $fetch2
